platform/Stream.cpp

```diff
@@ -23,7 +25,10 @@ Stream::Stream(const char *name) : FileLike(name), _file(NULL) {
     char buf[12]; /* :0x12345678 + null byte */
     std::sprintf(buf, ":%p", this);
     _file = std::fopen(buf, "w+");
-    mbed_set_unbuffered_stream(_file);
+    if (_file)
```

Comment:
Small nit if you're changing the file: We use squiggly brackets on single line if statement:
``` cpp
 if (_file) {
    blah
} else {
    bleh
}
```